# Business Understanding
## Overview of the business
A company that is seeking to expand into the aviation industry and wants buy aircrafts for its new division

## Business Problem

Being new in the aviation industry the company is unaware of the potential risks that are associated with aircrafts 

## Business Objectives
The company wants to understand the potential risks it may face in the aviation industry
The company wants to obtain a list of recommended aircrafts that are considered to have low risks in regards to its operational use

## Success Criteria
To obtain a list of aircrafts that are considered to have low risks

To outline the potential risks the company may experience in the aviation industry

To produce relevant visualisations that provide insights on aircraft and the aviation industry

## Project plan

To go through a dataset that contains accidents that have occurred in the aviation industry and to obtain relevant insights that can be able to solve the business problem






# Familization with the dataset

In [2]:
# Importing the pandas and matplotlib libraries
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
pd.options.display.max_columns = None

Loading the dataset to be used


In [4]:
aviation_data_df = pd.read_csv("data/AviationData.csv", encoding='latin-1', dtype='str')


In [5]:
usstate_codes_df = pd.read_csv("data/USState_Codes.csv")

Checking the head, tail and a sample of the dataset so to get an understanding of the dataset we are working with


In [6]:
aviation_data_df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
0          NaN        Fatal(2)       Destroyed               NaN   
1          NaN        Fatal(4)       Destroyed               NaN   
2          NaN        Fatal(3)       Destroyed               NaN   
3          NaN        Fatal(2)       Destroyed               NaN   
4          NaN        Fatal(1)       Destroyed               NaN   

  Registration.Number      Make     Model Amateur.Built Number.of.Engines  \
0              NC6404   Stinson     108-3            No                 1   
1              N5069P     Piper  PA24-180            No                 1   
2              N5142R    Cessna      172M            No                 1   
3              N1168J  Rockwell       112            No                 1   
4               N15NY    Cessna       501            No               NaN   

     Engine.Type FAR.Description Schedule Purpose.of.flight Air.carrier  \
0  Reciprocating             NaN      NaN          Personal         NaN   
1  Reciprocating             NaN      NaN          Personal         NaN   
2  Reciprocating             NaN      NaN          Personal         NaN   
3  Reciprocating             NaN      NaN          Personal         NaN   
4            NaN             NaN      NaN          Personal         NaN   

  Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                    2                      0                    0   
1                    4                      0                    0   
2                    3                    NaN                  NaN   
3                    2                      0                    0   
4                    1                      2                  NaN   

  Total.Uninjured Weather.Condition Broad.phase.of.flight   Report.Status  \
0               0               UNK                Cruise  Probable Cause   
1               0               UNK               Unknown  Probable Cause   
2             NaN               IMC                Cruise  Probable Cause   
3               0               IMC                Cruise  Probable Cause   
4               0               VMC              Approach  Probable Cause   

  Publication.Date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980

In [7]:
aviation_data_df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
88884  20221227106491           Accident      ERA23LA093  2022-12-26   
88885  20221227106494           Accident      ERA23LA095  2022-12-26   
88886  20221227106497           Accident      WPR23LA075  2022-12-26   
88887  20221227106498           Accident      WPR23LA076  2022-12-26   
88888  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
88884  Annapolis, MD  United States      NaN       NaN          NaN   
88885    Hampton, NH  United States      NaN       NaN          NaN   
88886     Payson, AZ  United States  341525N  1112021W          PAN   
88887     Morgan, UT  United States      NaN       NaN          NaN   
88888     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
88884          NaN           Minor             NaN               NaN   
88885          NaN             NaN             NaN               NaN   
88886       PAYSON       Non-Fatal     Substantial          Airplane   
88887          NaN             NaN             NaN               NaN   
88888          NaN           Minor             NaN               NaN   

      Registration.Number                        Make      Model  \
88884              N1867H                       PIPER  PA-28-151   
88885              N2895Z                    BELLANCA       7ECA   
88886              N749PJ  AMERICAN CHAMPION AIRCRAFT      8GCBC   
88887              N210CU                      CESSNA       210N   
88888              N9026P                       PIPER  PA-24-260   

      Amateur.Built Number.of.Engines Engine.Type FAR.Description Schedule  \
88884            No               NaN         NaN             091      NaN   
88885            No               NaN         NaN             NaN      NaN   
88886            No                 1         NaN             091      NaN   
88887            No               NaN         NaN             091      NaN   
88888            No               NaN         NaN             091      NaN   

      Purpose.of.flight         Air.carrier Total.Fatal.Injuries  \
88884          Personal                 NaN                    0   
88885               NaN                 NaN                    0   
88886          Personal                 NaN                    0   
88887          Personal  MC CESSNA 210N LLC                    0   
88888          Personal                 NaN                    0   

      Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
88884                      1                    0               0   
88885                      0                    0               0   
88886                      0                    0               1   
88887                      0                    0               0   
88888                      1                    0               1   

      Weather.Condition Broad.phase.of.flight Report.Status Publication.Date  
88884               NaN                   NaN           NaN       29-12-2022  
88885               NaN                   NaN           NaN              NaN  
88886               VMC                   NaN           NaN       27-12-2022  
88887               NaN                   NaN           NaN              NaN  
88888               NaN                   NaN           NaN       30-12-2022

In [8]:
aviation_data_df.sample(10)

Event.Id Investigation.Type Accident.Number  Event.Date  \
25991  20001212X23635           Accident      CHI90LA169  1990-07-08   
27670  20001212X16594           Accident      DEN91LA058  1991-03-31   
74169  20130904X10408           Incident      ENG13IA036  2013-08-31   
70946  20130620X30633           Accident      CEN12WA686  2011-10-14   
49035  20001212X21701           Accident      DCA00MA079  2000-08-08   
46543  20001212X19148           Accident      ANC99TA090  1999-07-08   
73852  20130707X93328           Accident      WPR13LA313  2013-07-07   
66653  20090630X85256           Accident      CEN09LA396  2009-06-29   
82926  20190311X61805           Accident      DCA19CA105  2019-03-11   
13771  20010110X00238           Accident      MKC86LA050  1986-01-12   

                         Location        Country Latitude Longitude  \
25991                  AURORA, IL  United States      NaN       NaN   
27670                LA JUNTA, CO  United States      NaN       NaN   
74169               Aguadilla, PR  United States  182941N  0067746W   
70946  Petit-Beauregard, Eswatini       Eswatini  465234N  0006355E   
49035              GREENSBORO, NC  United States      NaN       NaN   
46543               DEADHORSE, AK  United States      NaN       NaN   
73852              Parker Dam, CA  United States  341457N  1141016W   
66653             Shelbyville, IN  United States  039468N  0085502W   
82926                 Atlanta, GA  United States  333825N  0842539W   
13771              DES MOINES, IA  United States      NaN       NaN   

      Airport.Code                    Airport.Name Injury.Severity  \
25991          NaN                             NaN       Non-Fatal   
27670          LHX              LA JUNTA MUNICIPAL       Non-Fatal   
74169          BQN                RAFAEL HERNANDEZ             NaN   
70946          NaN                             NaN           Fatal   
49035          GSO    PIEDMONT TRIAD INTERNATIONAL       Non-Fatal   
46543          NaN                             NaN       Non-Fatal   
73852          NaN                             NaN           Fatal   
66653          NaN                             NaN       Non-Fatal   
82926         KATL  Hartsfield-Jackson Atlanta Int       Non-Fatal   
13771          DSM                 DES MOINES MUNI       Non-Fatal   

      Aircraft.damage Aircraft.Category Registration.Number  \
25991     Substantial               NaN              N33332   
27670     Substantial               NaN              N6904X   
74169           Minor          Airplane              PH-MCW   
70946       Destroyed          Airplane              G-BBEF   
49035     Substantial               NaN              N838AT   
46543     Substantial               NaN              N47122   
73852     Substantial      Weight-Shift              N467YN   
66653     Substantial          Airplane              N9085L   
82926           Minor          Airplane               N3756   
13771     Substantial               NaN              N8036B   

                                 Make         Model Amateur.Built  \
25991                           Piper      PA28-151            No   
27670                          Cessna          172B            No   
74169  BOEING COMPANY, LONG BEACH DIV  BOEING MD-11            No   
70946                           PIPER     PA-28-140            No   
49035                         Douglas       DC-9-32            No   
46543                            Bell          206B            No   
73852             NORTH WING UUM INC.      SPORT X2            No   
66653                        CHAMPION         7GCAA            No   
82926                          Boeing           737            No   
13771                          Cessna           172            No   

      Number.of.Engines    Engine.Type FAR.Description Schedule  \
25991                 1  Reciprocating             NaN      NaN   
27670                 1  Reciprocating             NaN      NaN   
74169       

In [9]:
aviation_data_df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [14]:
aviation_data_df.shape

(88889, 31)

In [10]:
len(aviation_data_df.columns)

31

In [11]:
usstate_codes_df.head()

US_State Abbreviation
0     Alabama           AL
1      Alaska           AK
2     Arizona           AZ
3    Arkansas           AR
4  California           CA

In [12]:
usstate_codes_df.tail()

US_State Abbreviation
57  Virgin Islands           VI
58   Washington_DC           DC
59  Gulf of mexico           GM
60  Atlantic ocean           AO
61   Pacific ocean           PO

In [16]:
usstate_codes_df.shape

(62, 2)

In [13]:
usstate_codes_df.columns

Index(['US_State', 'Abbreviation'], dtype='object')

# DATA UNDERSTANDING



The data folder contains two data sets

## Aviation dataset
This dataset contains data of reported aviation accidents and information regarding the accident reported.

The dataset contains 31 columns with each column containing information on the accident reported.

Some of the columns in the dataset include:date of the accident, the severity of the accident, country where the accident occurred, aircraft damage, aircraft type and model etc

The dataset contains 88889 records

## US State Codes dataset
This dataset contains information on the full name of abbreviated us state codes

The dataset contains two columns i.e the US state and its abbrevaiation

The dataset contains 62 records





# Data cleaning

## Aviation Dataset

In [17]:
aviation_data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Event.Id                88889 non-null  object
 1   Investigation.Type      88889 non-null  object
 2   Accident.Number         88889 non-null  object
 3   Event.Date              88889 non-null  object
 4   Location                88837 non-null  object
 5   Country                 88663 non-null  object
 6   Latitude                34382 non-null  object
 7   Longitude               34373 non-null  object
 8   Airport.Code            50249 non-null  object
 9   Airport.Name            52790 non-null  object
 10  Injury.Severity         87889 non-null  object
 11  Aircraft.damage         85695 non-null  object
 12  Aircraft.Category       32287 non-null  object
 13  Registration.Number     87572 non-null  object
 14  Make                    88826 non-null  object
 15  Mo

Using pd.info we can observe the following

a) every column contain string values

b)They are columns with null values

c)Columns names contain different structures and need to be standardized

### Converting column names to a simple standardized form

a)column names to be converted to lower case

b)column names to be seperated by underscore


In [19]:
aviation_data_df.columns.str.lower

<bound method _noarg_wrapper.<locals>.wrapper of <pandas.core.strings.StringMethods object at 0x000001AD8AD53AF0>>